bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/XMLValue.java

```diff
@@ -195,4 +197,20 @@ public void setChildren(XMLValue seq) {
     public abstract XMLValue children(String qname);
 
     public abstract XMLValue getItem(int index);
+
+    @Override
+    public void serialize(OutputStream outputStream) {
+        try {
+            if (outputStream instanceof BallerinaXMLSerializer) {
+                ((BallerinaXMLSerializer) outputStream).write(this);
+            } else {
```

Comment:
What would `outputStream` be if it comes to the else clause?

Reply:
https://github.com/ballerina-platform/ballerina-lang/blob/5a44cf7e3e4a992d7e2ae5fc68e74c85894457f3/bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BallerinaXMLSerializer.java#L153 here the outputStream is BallerinaXMLSerializer, which means we just want to append serialized xml to this outputStream.

Else it's a generic output stream that doesn't know how to write an xml value, so we decorate it with BallerinaXMLSerializer which knows how to write xml values.